server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -119,21 +125,78 @@ long createSecretSeries(
     r.setUpdatedat(now);
     r.setType(type);
     r.setRowHmac(rowHmac);
-    if (generationOptions != null) {
-      try {
-        r.setOptions(mapper.writeValueAsString(generationOptions));
-      } catch (JsonProcessingException e) {
-        // Serialization of a Map<String, String> can never fail.
-        throw Throwables.propagate(e);
-      }
-    } else {
-      r.setOptions("{}");
-    }
+    r.setOptions(getOptionsField(generationOptions));
+
     r.store();
 
     return r.getId();
   }
 
+  public long createDeletedSecretSeries(
+      String name,
+      Long ownerId,
+      String creator,
+      String description,
+      @Nullable String type,
+      @Nullable Map<String, String> generationOptions,
+      long now) {
+    long generatedId = rowHmacGenerator.getNextLongSecure();
```

Comment:
Curious why we aren't just using an  mysql auto-incremented for this?

Reply:
Ah, because we need to know the id before we add the `rowhmac`.

Reply:
oh yeah I had no idea why it was being done that way!